src/types/mod.rs

```diff
@@ -2650,3 +2650,143 @@ mod transaction_fee_distribution {
         }
     }
 }
+
+/// The network info of a node informs of the following:
+/// * The node id. An id which it uses to identify itself to other peers and it
+///   is used for logging purposes internally. NB. The 'node_id' is spoofable
+///   and as such should not serve as a trust instrument.
+/// * 'peer_total_sent' is the total amount of packets sent by the node.
+/// * 'peer_total_received' is the total amount of packets received by the node.
+/// * 'avg_bps_in' is the average bytes per second received by the node.
+/// * 'avg_bps_out' is the average bytes per second transmitted by the node.
+#[derive(Debug)]
+pub struct NetworkInfo {
+    pub node_id:             String,
+    pub peer_total_sent:     u64,
+    pub peer_total_received: u64,
+    pub avg_bps_in:          u64,
+    pub avg_bps_out:         u64,
+}
+
+/// A node can either be a 'Bootstrapper' or 'Node'.
+/// The 'Bootstrapper' is a special kind of node that does not participate
+/// in consensus but merely relay peers.
+/// The 'Node' is a node that participates in consensus either passively (only
+/// processing incoming blocks) or actively i.e. by being a baker.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Debug, Display, From, TryInto)]
+pub enum NodeType {
+    Bootstrapper,
+    Node,
+}
+
+impl TryFrom<i32> for NodeType {
+    type Error = anyhow::Error;
+
+    // Parse the node type.
+    // The node type is specified via an enum:
+    // * 0 => The node is a bootstrapper and is simply relaying peers.
+    // * 1 => The node is participating in consensus either actively (baking) or
+    //   passively (solely processing blocks).
+    // If the result is none of the above we return an error "Malformed node type."
+    fn try_from(v: i32) -> Result<Self, Self::Error> {
+        match v {
+            x if x == NodeType::Bootstrapper as i32 => Ok(NodeType::Bootstrapper),
+            x if x == NodeType::Node as i32 => Ok(NodeType::Node),
+            _ => Err(anyhow::anyhow!("Malformed node type")),
+        }
+    }
+}
+
+#[derive(Debug)]
+/// Consensus related information for a node.
+pub enum ConsensusInfo {
```

Comment:
We already have a type `ConsensusInfo` I think, with a very different meaning. And it's a struct.

Reply:
Yes. Renamed to `NodeConsensusStatus`.

Reply:
Renamed it.